File: Languages/Dart/task_list.dart
import 'dart:io';

void main() {
  List<String> tasks = [];
  while (true) {
    print('\nMenu:');
    print('1. Adicionar Tarefa');
    print('2. Remover Tarefa');
    print('3. Mostrar Tarefas');
    print('4. Sair');
    
    String? choice = stdin.readLineSync();
    
    if (choice == '1') {
      print('Digite a tarefa:');
      String? task = stdin.readLineSync();
      if (task != null && task.isNotEmpty) {
        tasks.add(task);
        print('Tarefa adicionada.');
      } else {
        print('Tarefa não pode ser vazia.');
      }
    } else if (choice == '2') {
      print('Tarefas:');
      for (int i = 0; i < tasks.length; i++) {
        print('${i + 1}. ${tasks[i]}');
      }
      print('Digite o número da tarefa a ser removida:');
      String? indexInput = stdin.readLineSync();
      if (indexInput != null) {
        int index = int.tryParse(indexInput) ?? -1;
        if (index > 0 && index <= tasks.length) {
          String removedTask = tasks.removeAt(index - 1);
          print('Tarefa "$removedTask" removida.');
        } else {
          print('Índice inválido.');
        }
      }
    } else if (choice == '3') {
      print('Tarefas:');
      if (tasks.isEmpty) {
        print('Nenhuma tarefa na lista.');
      } else {
        for (int i = 0; i < tasks.length; i++) {
          print('${i + 1}. ${tasks[i]}');
        }
      }
    } else if (choice == '4') {
      print('Saindo...');
      break;
    } else {
      print('Opção inválida. Tente novamente.');
    }
  }
}

File: Languages/Go/main/task_list.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var tasks []string
	var choice string

	for {
		fmt.Println("\nMenu:")
		fmt.Println("1. Adicionar Tarefa")
		fmt.Println("2. Remover Tarefa")
		fmt.Println("3. Mostrar Tarefas")
		fmt.Println("4. Sair")
		fmt.Print("Escolha uma opção: ")
		fmt.Scanln(&choice)

		switch choice {
		case "1":
			var task string
			fmt.Print("Digite a tarefa: ")
			fmt.Scanln(&task)
			if strings.TrimSpace(task) != "" {
				tasks = append(tasks, task)
				fmt.Println("Tarefa adicionada.")
			} else {
				fmt.Println("A tarefa não pode ser vazia.")
			}
		case "2":
			if len(tasks) == 0 {
				fmt.Println("Nenhuma tarefa para remover.")
				continue
			}
			fmt.Println("Tarefas:")
			for i, t := range tasks {
				fmt.Printf("%d. %s\n", i+1, t)
			}
			var index int
			fmt.Print("Digite o número da tarefa a ser removida: ")
			fmt.Scanln(&index)
			if index > 0 && index <= len(tasks) {
				removedTask := tasks[index-1]
				tasks = append(tasks[:index-1], tasks[index:]...)
				fmt.Printf("Tarefa \"%s\" removida.\n", removedTask)
			} else {
				fmt.Println("Índice inválido.")
			}
		case "3":
			if len(tasks) == 0 {
				fmt.Println("Nenhuma tarefa na lista.")
			} else {
				fmt.Println("Tarefas:")
				for i, t := range tasks {
					fmt.Printf("%d. %s\n", i+1, t)
				}
			}
		case "4":
			fmt.Println("Saindo...")
			return
		default:
			fmt.Println("Opção inválida. Tente novamente.")
		}
	}
}